apps/prairielearn/src/question-servers/index.ts

```diff
@@ -3,14 +3,56 @@ import { Question, Course, Variant, Submission, CourseInstance } from '../lib/db
 export type QuestionType = Question['type'];
 export type EffectiveQuestionType = 'Calculation' | 'Freeform';
 
-type QuestionServerCallback = (err: Error | null, courseIssues: Error[], data: any) => void;
+type QuestionServerCallback<T> = (err: Error | null, courseIssues: Error[], data: T) => void;
+
+export interface GenerateResultData {
+  params: Record<string, any>;
+  true_answer: Record<string, any>;
+  options?: Record<string, any> | null;
```

Comment:
This is a value that is read by the question-variant.js library, but not updated by the freeform question server. I am not familiar with the v2 question server code, so I assume this is a value created there, but I'd welcome a confirmation from @mwest1066 or @nwalters512 if this is indeed the case.

Reply:
From what I can tell, v2 questions supported an `options` property in their JSON file. Here's the code that reads/validates it at sync time:

https://github.com/PrairieLearn/PrairieLearn/blob/85e938e1e51207eeba4557e31dde20ef502bc934/apps/prairielearn/src/sync/course-db.js#L1021

I believe your types here are correct in that case.